urwid/main_loop.py

```diff
@@ -1468,6 +1468,8 @@ def _exception_handler(self, loop, context):
                 # Store the exc_info so we can re-raise after the loop stops
                 import sys
                 self._exc_info = sys.exc_info()
+                if self._exc_info == (None, None, None):
+                    self._exc_info = exc
```

Comment:
You can remove the need for the [`run` changes](https://github.com/urwid/urwid/pull/283/files#diff-f0d3f08367104b70907965026fd336e3R1486) by doing this instead:

```python
self._exc_info = (type(exc), exc, None)
```

Reply:
That's much cleaner, thanks!